src/ctorch.h

```diff
@@ -38,7 +38,12 @@ typedef enum {
 } torch_data_t;
 
 // Device types
-typedef enum { torch_kCPU, torch_kCUDA } torch_device_t;
+typedef enum {
```

Comment:
Consider adding note to say these are defined in CMakeLists as Preprocessor variables?
```suggestion
// Defined in main CMakeLists and passed via preprocessor
typedef enum {
```